pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -814,6 +814,71 @@ ssize_t sock_dtls_recv_aux(sock_dtls_t *sock, sock_dtls_session_t *remote,
     }
 }
 
+ssize_t sock_dtls_recv_buf_aux(sock_dtls_t *sock, sock_dtls_session_t *remote,
+                               void **data, void **buf_ctx, uint32_t timeout,
+                               sock_dtls_aux_rx_t *aux)
+{
+    assert(sock);
+    assert(data);
+    assert(buf_ctx);
+    assert(remote);
+
+    sock_udp_ep_t ep;
+
+    /* 2nd call to the function (with ctx set) will free the data */
+    if (*buf_ctx) {
+        int res = sock_udp_recv_buf_aux(sock->udp_sock, data, buf_ctx,
+                                        timeout, &ep, (sock_udp_aux_rx_t *)aux);
+        assert(res == 0);
+        return res;
+    }
```

Comment:
I could not find this usage pattern (of `sock_udp_recv_buf_aux` to "free" the `ctx`and `data`) at another point, but i think if it is just a "free" call it should have a timeout of 0. (i might be wrong).

(I am also not sure why there is the assertion that there will be no data to be received (that therefor would need to be processed)

Reply:
That's also done in [nanocoap_sock](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/application_layer/nanocoap/sock.c#L173).

IMHO the `sock_udp_recv_buf_aux()` is just a bit clunky. If we would need to do application level re-assembly, that would void the point of the function as we would need a 2nd buffer for re-assembly again.

Reply:
seems like the documentation for `sock_udp_recv_buf_aux` and `sock_udp_recv_buf` need an update and/or the ctx thing removed (out of scope for this PR)